File: 5-caketransport/solution.py
import sys

def scan_cake(n):
    mnx = sys.maxsize
    mxx = -sys.maxsize

    xs = [int(x) for x in input().split()]
    ys = [int(x) for x in input().split()]
    for x in xs:
        assert abs(x) <= 1_000_000_000
        mnx = min(mnx, x)
        mxx = max(mxx, x)
    for y in ys:
        assert abs(y) <= 1_000_000_000
    return (mnx, mxx)


n, m = input().split()
n = int(n)
m = int(m)

cake1 = scan_cake(n)
cake2 = scan_cake(m)

if max(cake1[0], cake2[0]) > min(cake1[1], cake2[1]):
    print("No Collision!")
else:
    print("Collision!")

File: 5-caketransport/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int n, m, tx, ty, min1 = 1e9, max1 = -1e9, min2 = 1e9, max2 = -1e9; 

int main() {
    scanf("%d %d", &n, &m);

    for (int i = 1; i <= n; i++) {
        scanf("%d", &tx);
        min1 = min(min1, tx);
        max1 = max(max1, tx);
    }
    for (int i = 1; i <= n; i++) {
        scanf("%d", &ty);
    }

    for (int i = 1; i <= m; i++) {
        scanf("%d", &tx);
        min2 = min(min2, tx);
        max2 = max(max2, tx);
    }
    for (int i = 1; i <= n; i++) {
        scanf("%d", &ty);
    }

    if (max(min1, min2) <= min(max1, max2)) {
        printf("Collision!\n");
    } else {
        printf("No Collision!\n");
    }
}
